src/main/java/core/basesyntax/services/impl/OperationProcessorImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.services.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.services.OperationProcessor;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.List;
+import java.util.Map;
+
+public class OperationProcessorImpl implements OperationProcessor {
+    private final Map<FruitTransaction.Operation, OperationHandler> processedOperations;
+
+    public OperationProcessorImpl(
+            Map<FruitTransaction.Operation, OperationHandler> processedOperations) {
+        this.processedOperations = processedOperations;
+    }
+
+    @Override
+    public void manageTransactions(List<FruitTransaction> transactions) {
+        transactions.forEach(t -> processedOperations.get(t.getOperation()).calculateOperation(t));
```

Comment:
Ensure that the `processedOperations` map contains handlers for all possible operations to avoid potential `NullPointerException`. Consider adding a check or a default handler to handle cases where an operation might not be present in the map.